### PR TITLE
Jenkinsfile: Post results on PR

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
Gets metadata from build server.
Update env script to set repo_name for metadata.
Compiling results script only uses metadata and does not generate.
Add release as build parameter for RIOT.
Post test results and link on robot PRs.
Always notify at finish.

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
It seems that the compile results script doesn't work anymore since it is run on master and master does not have RIOT included...

Maybe it is better to have these scripts somewhere else, either way it is a TODO to fix...

**Comment by MrKevinWeiss** [comment]
We should also show the errors, the only problem is if it is empty then it fails...

**Comment by MrKevinWeiss** [comment]
THISISA TEST
and2

**Comment by MrKevinWeiss** [comment]
Take a look at the pr 89  tests and nightly tests to verify everything is OK.

Make sure the robot.xml contains the proper commit id of both robotfw and riot in the metadata tag.

**Comment by MrKevinWeiss** [comment]
nightlies only update if everything is on master

File: Jenkinsfile [hunk 1]
```diff
@@ -143,25 +143,38 @@ def stepGetRiotUrlAndCommit() {
...
                 env.GIT_COMMIT]
     }
     else if (ver == 'submodule') {
+        sh script: "echo ${repo_name} using submodule",
+            label: "echo ${repo_name} using submodule"
         return getUrlAndCommitOfSubmodule("RIOT-OS", "RobotFW-Tests")
     }
+    else if (ver == 'release') {
+        sh script: "echo ${repo_name} using release ${ver_arg}",
+            label: "echo ${repo_name} using release ${ver_arg}"
```
Comment (pokgak) [inline Jenkinsfile:175]:
```suggestion
            label: "echo ${repo_name} using release ${ver_arg}"
        assert ver_arg != '0' : "Selected release of ${repo_name} but no release version entered"
```

**Review by pokgak — APPROVED** [review]
There are some `No such saved stash ‘saml10-xpro_tests_periph_uart’` errors in PR 89 result in HIL. Is that normal?

Other than that and the comments below, everything else looks okay.

File: Jenkinsfile (hunk 1 above)
Comment (MrKevinWeiss) [reply Jenkinsfile:175]: I don't think it is worth rerunning tests for this change, it should fail if nothing is found (the API and parsing would crash it)

**Comment by MrKevinWeiss** [comment]
> There are some `No such saved stash ‘saml10-xpro_tests_periph_uart’` errors in PR 89 result in HIL. Is that normal?

Yup, I have not found a way to test if a stash is present.  Currently if fails in a decent way.  Maybe later we would keep track of all the jobs or something, still learning about how the map works...

**Comment by MrKevinWeiss** [comment]
@pokgak Thanks for looking at this so late! Great that we can get it in before the next nightly test.

**Pull request merged**